samples/net/cloud/aws_mqtt/src/main.c

```diff
@@ -0,0 +1,310 @@
+				LOG_ERR("MQTT PUBACK error %d", evt->result);
+				break;
+			}
+
+			LOG_INF("PUBACK packet id: %u", evt->param.puback.message_id);
+
+			break;
+
+		case MQTT_EVT_PUBREC:
+			if (evt->result != 0) {
+				LOG_ERR("MQTT PUBREC error %d", evt->result);
+				break;
+			}
+
+			LOG_INF("PUBREC packet id: %u", evt->param.pubrec.message_id);
+
+			const struct mqtt_pubrel_param rel_param = {
+				.message_id = evt->param.pubrec.message_id
+			};
+			err = mqtt_publish_qos2_release(client, &rel_param);
+			if (err != 0) {
+				LOG_ERR("Failed to send MQTT PUBREL: %d", err);
+			}
+
+			break;
+
+		case MQTT_EVT_PUBCOMP:
+			if (evt->result != 0) {
+				LOG_ERR("MQTT PUBCOMP error %d", evt->result);
+				break;
+			}
+
+			LOG_INF("PUBCOMP packet id: %u",
+					evt->param.pubcomp.message_id);
+
+			break;
+
+		default:
+			break;
+	}
+}
+
+static void client_init(struct mqtt_client *client)
+{
+	mqtt_client_init(client);
+	broker_init();
+
+	client->password = NULL;
+	client->user_name = NULL;
+
+	/* MQTT client configuration */
+	client->broker = &broker;
+	client->evt_cb = mqtt_evt_handler;
+	client->client_id.utf8 = (u8_t *)MQTT_CLIENTID;
+	client->client_id.size = strlen(MQTT_CLIENTID);
+	client->protocol_version = MQTT_VERSION_3_1_1;
+
+	/* MQTT buffers configuration */
+	client->rx_buf = rx_buffer;
+	client->rx_buf_size = sizeof(rx_buffer);
+	client->tx_buf = tx_buffer;
+	client->tx_buf_size = sizeof(tx_buffer);
+
+	client->transport.type = MQTT_TRANSPORT_SECURE;
+	struct mqtt_sec_config *tls_config = &client->transport.tls.config;
+	tls_config->peer_verify = TLS_PEER_VERIFY_REQUIRED;
+	tls_config->cipher_list = NULL;
+	tls_config->sec_tag_list = m_sec_tags;
+	tls_config->sec_tag_count = ARRAY_SIZE(m_sec_tags);
+	tls_config->hostname = CONFIG_AWS_HOSTNAME;
+	client->transport.type = MQTT_TRANSPORT_SECURE;
+}
+
+static int wait_for_input(int timeout)
+{
+        int res;
+        struct zsock_pollfd fds[1] = {
+                [0] = {.fd = client_ctx.transport.tls.sock,
+                      .events = ZSOCK_POLLIN,
+                      .revents = 0},
+        };
+
+        res = zsock_poll(fds, 1, timeout);
+        if (res < 0) {
+                LOG_ERR("poll read event error");
+                return -errno;
+        }
+        return res;
+}
+
+static int try_to_connect(struct mqtt_client *client)
+{
+	int rc;
+
+	LOG_DBG("attempting to connect...");
+
+	client_init(client);
+	rc = mqtt_connect(client);
+	if (rc != 0) {
+		PRINT_RESULT("mqtt_connect", rc);
+	}else {
+		if (wait_for_input(5000) > 0) {
+			mqtt_input(client);
+			if (!connected) {
+				LOG_INF("Aborting connection\n");
+				mqtt_abort(client);
+			}
+		}
+	}
+	if (connected) {
+		return 0;
+	}
+	return -EINVAL;
+}
+
+static void publish(struct mqtt_client *client, enum mqtt_qos qos)
+{
+        char payload[] = "{id=123}";
+        char topic[] = "/things/";
+        u8_t len = strlen(topic);
+        struct mqtt_publish_param param;
+	char pub_msg[64];
```

Comment:
Indentation wrong. Also we try to avoid magic constants, please use defines instead for the length